drivers/hmis/app/graphql/mutations/update_form_definition.rb

```diff
@@ -16,11 +18,10 @@ def resolve(id:, input:)
       raise 'not found' unless definition
       raise 'not allowed' if definition.managed_in_version_control?
 
-      # The UI currently does allow changing the form role, so we make this permission check twice:
-      # - to confirm the user has permission to manage the role from the input (in case it's changing)
-      # - to confirm the user has permission to manage forms for the original role (from the definition)
+      access_denied! unless policy_for(definition, policy_type: :form_definition).can_manage_form?
+
+      # If the form role has been changed, confirm the user also has permission for the new role.
       access_denied! if input.role && !current_user.can_manage_forms_for_role?(input.role)
```

Comment:
perhaps this additional check could be rolled up to the policy:
```ruby 
   class Hmis::AuthPolicies::FormDefinitionPolicy < Hmis::AuthPolicies::BasePolicy
      def can_update_form?(new_role: nil)
        return false unless can_manage_form?
        return true if new_role.blank?
        user.can_manage_forms_for_role?(new_role)
      end
    end
```
then
`access_denied! unless policy_for(definition, policy_type: :form_definition)
      .can_update_form?(new_role: input.role)`